internal/provider/cluster_resource.go

```diff
@@ -882,6 +906,13 @@ func loadClusterToTerraformState(
 		state.ParentId = types.StringValue(*clusterObj.ParentId)
 	}
 
+	if clusterObj.DeleteProtection != nil &&
```

Comment:
I forget, was DeleteProtection being a pointer just because we missed adding it to that json spec array?

Reply:
Yes, but I think it's actually the thing we want since we don't include the UNSPECIFIED value in the JSON spec so we'd have no way to represent an unset value without this being a pointer.

Reply:
Interesting. I feel like we'd want to default the UNSPECIFIED value to false.

Reply:
The server should never return `UNSPECIFIED`, but the client can leave it unset which will map to `UNSPECIFIED` on the server, which the server will map to `DISABLED` during cluster creation. Leaving it unset during update will map to `UNSPECIFIED` on the server which will mean "don't change the value".

Reply:
I guess i'm surprised we would allow explicitly passing a value that meant don't change it via a patch when they should already be able to exclude passing it in.

Reply:
I think it's kind of a dissonance between our different JSON, protobuf, and various client and server generated code. `UNSPECIFIED` isn't included in the spec so users should never know that they can pass it, so the way they'll represent "unset" is to not set the pointer in the Go SDK or leave out the field from the JSON payload.

In the past there has been value in having a way to represent both "unset the value" and "leave the value alone" which I think can make it valuable to have the "`UNSPECIFIED'" value even if the only the server knows about it for now.

Reply:
Since we don't generally publish it as part of the spec, I guess its probably not causing any confusion.  incidentally, I do see it in a few of the enums in the api spec.  Not this particular one.  Do you have any examples of times this has provided value for the api?

Reply:
Looks like this is why there are some `"UNSPECIFIED"` values in the spec, they use a different name for the default `0` value. Not ideal...
![Screenshot 2024-04-29 at 1 06 14 PM](https://github.com/cockroachdb/terraform-provider-cockroach/assets/5905206/6a9f6221-f138-4e6f-8b63-47a9a184694c)

I don't know if I have an example where an enum in particular needed the "unset" vs "don't touch" property but it came up when we were implementing the API for serverless cluster usage limits and we had to distinguish between "leave the usage limits alone", "set usage limits to unlimited" and "set usage limits to 0".

In general I just find that it's good for backwards compatibility and nearly impossible to retrofit on later.